### ISSUE TITLE
Graph convolutional network module for text

### ISSUE THREAD

**Issue by dchang56** [allenai/allennlp#2739]
**Is your feature request related to a problem? Please describe.**
In the past couple of years there's been an increase in application of graph convolutional networks in NLP, and it seems promising and interesting enough that it could be good to have a module for it to give researchers the option of incorporating it into their models.

**Describe the solution you'd like**
A module that takes in a graph data structure (adjacency matrix, node/edge features, etc) and outputs updated graph data structure, and preferably be compatible with graph network implementations from Pytorch Geometric

**Comment by DeNeutoy** [comment]
This is not an area we've been working on, but it should be straightforward to implement a `SparseFormatField` in one of the more common sparse matrix formats. This could then be compatible with some GCN type encoders, such as those in Pytorch Geometric. It's somewhat unlikely the allennlp team will be working on this in the near future, but contributions or notifications of separate implementations where people can go to find GCN type support would be welcome!

**Comment by dchang56** [comment]
Thank you, I'll keep an eye out and try to implement it in the near future.
Any and all suggestions/ideas for implementing graph convolution within the AllenNLP framework would be greatly appreciated :)

**Comment by matt-gardner** [comment]
I know of some folks at Tel Aviv using GCNs inside of AllenNLP for some applications, though the paper using it is under review, so they haven't submitted code upstream.  I'm not sure whether to recommend waiting for them or not.  Just FYI.

**Comment by dchang56** [comment]
@matt-gardner That's great to hear! Do you think it would be possible to put me in contact with them if it's not too much trouble?

**Comment by matt-gardner** [comment]
@dchang56, I sent an email to the folks who were working on this, and they may or may not comment here; we'll see.

**Comment by benbogin** [comment]
@dchang56 @matt-gardner Hey! Not sure if this is helpful, but actually in our implementation we simply use the following gist:
https://github.com/pcyin/pytorch-gated-graph-neural-network/blob/master/gnn.py
which implements [Gated Graph Sequence Neural Networks](https://arxiv.org/pdf/1511.05493.pdf). We pass it initial embeddings for each node, a list of adjacency pairs and the number of timesteps, and it outputs the updated encoding for each node, which was sufficient for our needs.

**Comment by DeNeutoy** [comment]
Closing, we probably shouldn't support this directly.

**Comment by EazyReal** [comment]
> I know of some folks at Tel Aviv using GCNs inside of AllenNLP for some applications, though the paper using it is under review, so they haven't submitted code upstream. I'm not sure whether to recommend waiting for them or not. Just FYI.

@matt-gardner I am working on using GNN with allennlp too, is there a way I can contact them? Thanks a lot!

**Comment by matt-gardner** [comment]
@EazyReal, you can see the links above, and also check out [this repo](https://github.com/benbogin/spider-schema-gnn-global).

**Comment by EazyReal** [comment]
@matt-gardner ,
Thanks for your reply!
I think I did not express my problem well enough, the point is that I want to use mini-batch on graphs (like in Pytorch Geometric, allennlp provides only (Dense)AdjacencyField).
It seems that the implementation of the link (https://github.com/pcyin/pytorch-gated-graph-neural-network/blob/master/gnn.py) does not support batch training.
And the repo you mentioned uses KnowledgeGraphField, which is not supported in allennlp 1.0.
So I actually coded a SparseadjacencyField yesterday and it seems working well now.
![image](https://user-images.githubusercontent.com/8047065/90599450-769faf80-e227-11ea-8399-bdb767dfe8c2.png)
![image](https://user-images.githubusercontent.com/8047065/90599377-54a62d00-e227-11ea-8994-107f1c814766.png)
